docs/how-to/service-dependencies.md

```diff
@@ -1,11 +1,270 @@
+    requires:
+      - backend
+  backend:
+    override: replace
+    command: python3 -m http.server 8081
+    startup: enabled
+    requires:
+      - database
+  database:
+    override: replace
+    command: python3 -m http.server 3306
+    startup: enabled
+```
+
+In the layer configuration above, the ``backend`` service is dependent on the
+``database`` service, and the ``frontend`` service is dependent on the
+``backend`` service.
+
+### Specify start order for dependent services
+
+To specify the order in which one or more dependent services must start
+successfully, relative to a given service, use the ``before`` or ``after`` keys.
+
+```{code-block} yaml
+   :emphasize-lines: 8, 9, 16, 17
+
+services:
+  frontend:
+    override: replace
+    command: python3 -m http.server 8080
+    startup: enabled
+    requires:
+      - backend
+    after:
+      - backend
+  backend:
+    override: replace
+    command: python3 -m http.server 8081
+    startup: enabled
+    requires:
+      - database
+    after:
+      - database
+  database:
+    override: replace
+    command: python3 -m http.server 3306
+    startup: enabled
+```
+
+In the updated layer above, the ``backend`` service requires the ``database``
+service to be started before it, and the ``frontend`` service requires the
+``backend`` service to be started before it.
+
+```{note}
+Currently, ``before`` and ``after`` are of limited usefulness, because Pebble only waits 1 second before moving on to start the next service, with no additional checks that the previous service is operating correctly.
 
 If the configuration of `requires`, `before`, and `after` for a service results in a cycle or "loop", an error will be returned when attempting to start or stop the service.
+```
+
+% Does it only check that the status is "Active"? So what does `require` do?
+
+## Verify service dependencies
+
+To verify the start order of services, start the Pebble daemon again:
+
+```{terminal}
+   :input: pebble run
+
+2024-06-28T03:53:23.307Z [pebble] Started daemon.
+[database backend frontend]
+2024-06-28T03:53:23.313Z [pebble] POST /v1/services 3.103291ms 202
+2024-06-28T03:53:23.317Z [pebble] Service "database" starting: python3 -m http.server 3306
+2024-06-28T03:53:24.324Z [pebble] Service "backend" starting: python3 -m http.server 8081
+2024-06-28T03:53:25.332Z [pebble] Service "frontend" starting: python3 -m http.server 8080
+2024-06-28T03:53:26.337Z [pebble] GET /v1/changes/1/wait 3.023563626s 200
+2024-06-28T03:53:26.338Z [pebble] Started default services with change 1.
+```
+
+The start order is now ``database`` -> ``backend`` -> ``frontend`` and all
+services are "active".
+
+```{terminal}
+   :input: pebble services
+   
+Service   Startup  Current  Since
+backend   enabled  active   today at 10:17 CST
+database  enabled  active   today at 10:17 CST
+frontend  enabled  active   today at 10:17 CST
+```
+
+To further verify the service dependencies, we can force fail a service that is
```

Comment:
"Force fail" doesn't mean much to me. What about "if we force a service that is requires by another service to fail"? And how is it being forced to fail here -- that's unclear.

Reply:
Good point.

Reply:
Hey @benhoyt , 

"How is it being forced to fail here." For the first part it is a general guide on how you may want to test this. So I added (back) more context for the web app example we are using here.

Hope the changes make sense.
---

To further verify the service dependencies, we force a service that is required
by another service to fail.

In this example, if we force the `database` service to fail (by using port 3306
for another process), the output for the `pebble run` command should be similar
to: